### ISSUE TITLE
Port planck.core & planck.io to Replete

### ISSUE THREAD

**Issue by roman01la** [replete-repl/replete-shared#4]
I think it'd be nice to have at least basic network IO in Replete. @mfikes what parts of https://github.com/planck-repl/planck#clojurecore---planckcore do you think makes sense to port from Planck?

**Comment by mfikes** [comment]
We could just copy it all _en masse_, porting the lower level functions on each device, maybe renaming namespaces to `replete.core` and `replete.io`, _etc._ followed by deleting any things we don't want to support.

Thinking about file IO in particular, it we could perhaps arrange it so that file IO goes to some directory on local to the app? That way, everything "just works", and since it mirrors Clojure's patterns, is really useful for people wanting to experiment and learn how to do IO with Clojure.

We could even port the `planck.http` namespace so that things like this just work:

```clojure
(require '[replete.io :as io])
(io/copy (io/reader "http://planck-repl.org") (io/file "page.html"))
```

```clojure
(require '[replete.core :refer [slurp]])
(print (slurp "page.html")
```

This also makes it so that programs written on Replete, Planck, Clojure (and perhaps even Lumo with [abio-node](https://github.com/abiocljs/abio-node)) are generally movable between those environments.

**Comment by roman01la** [comment]
Sounds good. I think by default on Android file IO goes into app specific directory named after app's package name.

I'll prepare a WIP PR for this.

**Comment by roman01la** [comment]
Here's a complete list of "hooks" that we have for now. We could mark which ones to keep for Replete.

- [ ] `REPLETE_LOAD_FROM_JAR`
- [x] `REPLETE_LOAD`
- [x] `REPLETE_FILE_READER_OPEN`
- [x] `REPLETE_FILE_READER_CLOSE`
- [x] `REPLETE_FILE_READER_READ`
- [x] `REPLETE_FILE_WRITER_OPEN`
- [x] `REPLETE_FILE_WRITER_WRITE`
- [x] `REPLETE_FILE_WRITER_FLUSH`
- [x] `REPLETE_FILE_WRITER_CLOSE`
- [x] `REPLETE_FILE_INPUT_STREAM_OPEN`
- [x] `REPLETE_FILE_INPUT_STREAM_READ`
- [x] `REPLETE_FILE_INPUT_STREAM_CLOSE`
- [x] `REPLETE_FILE_OUTPUT_STREAM_OPEN`
- [x] `REPLETE_FILE_OUTPUT_STREAM_WRITE`
- [x] `REPLETE_FILE_OUTPUT_STREAM_FLUSH`
- [x] `REPLETE_FILE_OUTPUT_STREAM_CLOSE`
- [x] `REPLETE_FSTAT`
- [x] `REPLETE_DELETE`
- [x] `REPLETE_IS_DIRECTORY`
- [x] `REPLETE_LIST_FILES`
- [x] `REPLETE_MKDIRS`
- [x] `REPLETE_COPY`
- [x] `REPLETE_REQUEST`
- [ ] `REPLETE_EXIT_WITH_VALUE`
- [ ] `REPLETE_RAW_READ_STDIN`
- [x] `REPLETE_RAW_WRITE_STDOUT`
- [x] `REPLETE_RAW_FLUSH_STDOUT`
- [x] `REPLETE_RAW_WRITE_STDERR`
- [x] `REPLETE_RAW_FLUSH_STDERR`
- [ ] `REPLETE_INITIAL_COMMAND_LINE_ARGS`
- [ ] `REPLETE_READ_PASSWORD`
- [x] `REPLETE_SLEEP`

**Comment by mfikes** [comment]
Cool. I'll port the same to Replete iOS. Here is a progress list of the [port](https://github.com/replete-repl/replete-ios/tree/io-hooks) for that target:

- [ ] `REPLETE_LOAD_FROM_JAR`
- [ ] `REPLETE_LOAD`
- [x] `REPLETE_FILE_READER_OPEN`
- [x] `REPLETE_FILE_READER_CLOSE`
- [x] `REPLETE_FILE_READER_READ`
- [x] `REPLETE_FILE_WRITER_OPEN`
- [x] `REPLETE_FILE_WRITER_WRITE`
- [x] `REPLETE_FILE_WRITER_FLUSH`
- [x] `REPLETE_FILE_WRITER_CLOSE`
- [x] `REPLETE_FILE_INPUT_STREAM_OPEN`
- [x] `REPLETE_FILE_INPUT_STREAM_READ`
- [x] `REPLETE_FILE_INPUT_STREAM_CLOSE`
- [x] `REPLETE_FILE_OUTPUT_STREAM_OPEN`
- [x] `REPLETE_FILE_OUTPUT_STREAM_WRITE`
- [x] `REPLETE_FILE_OUTPUT_STREAM_FLUSH`
- [x] `REPLETE_FILE_OUTPUT_STREAM_CLOSE`
- [x] `REPLETE_FSTAT`
- [x] `REPLETE_DELETE`
- [x] `REPLETE_IS_DIRECTORY`
- [x] `REPLETE_LIST_FILES`
- [x] `REPLETE_MKDIRS`
- [x] `REPLETE_COPY`
- [x] `REPLETE_REQUEST`
- [ ] `REPLETE_EXIT_WITH_VALUE`
- [ ] `REPLETE_RAW_READ_STDIN`
- [x] `REPLETE_RAW_WRITE_STDOUT`
- [x] `REPLETE_RAW_FLUSH_STDOUT`
- [x] `REPLETE_RAW_WRITE_STDERR`
- [x] `REPLETE_RAW_FLUSH_STDERR`
- [ ] `REPLETE_INITIAL_COMMAND_LINE_ARGS`
- [ ] `REPLETE_READ_PASSWORD`
- [x] `REPLETE_SLEEP`

**Comment by mfikes** [comment]
I think Replete iOS is essentially ready with this stuff (certainly ready for beta testing), via https://github.com/replete-repl/replete-ios/pull/123

**Comment by roman01la** [comment]
@mfikes That's great! I'm missing two more hooks on Android side, but I think iOS can release it already, since we still need to test the whole Android app first.

**Comment by mfikes** [comment]
@roman01la Cool. Yeah, we could get this stuff in, and perhaps put the Android and iOS apps into betas. (I can even put the Android app into the Google Play store in a beta mode if that helps / works.)

**Comment by roman01la** [comment]
I'm done with IO hooks. `FSTAT` on Android doesn't seem to provide much. For now it's only type directory or file and modified date.

**Comment by roman01la** [comment]
resolved by #5
